lib/libutee/arch/arm/ftrace/ftrace.c

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (C) 2019, Linaro Limited
+ */
+
+#include <string.h>
+#include <user_ta_header.h>
+#include <utee_syscalls.h>
+
+#define FTRACE_RETFUNC_DEPTH		50
+
+static unsigned long ftrace_ret_stack[FTRACE_RETFUNC_DEPTH];
+static unsigned int ftrace_ret_idx;
+static const char hex_str[] = "0123456789abcdef";
+
+void __ftrace_return(void);
+void ftrace_enter(unsigned long pc, unsigned long *lr);
+
+static void __noprof ftrace_memmove(struct ftrace_buf *fbuf, size_t size)
+{
+	char *dst = (char *)fbuf + fbuf->buf_off;
+	const char *src = (char *)fbuf + fbuf->buf_off + size;
+	size_t len = 0;
+
+	fbuf->curr_size -= size;
+	len = fbuf->curr_size;
+
+	while (len--)
+		*dst++ = *src++;
+}
+
+static size_t __noprof to_func_enter_fmt(char *buf, unsigned long pc)
+{
+	char *str = buf;
+	uint32_t i, addr_size = 2 * sizeof(unsigned long);
+
+	for (i = 0; i < ftrace_ret_idx; i++)
+		*str++ = ' ';
+
+	*str++ = '0';
+	*str++ = 'x';
+
+	for (i = 0; i < addr_size; i++)
+		*str++ = hex_str[(pc >> 4*(addr_size - i - 1)) & 0xf];
+
+	*str++ = '(';
+	*str++ = ')';
+	*str++ = ' ';
+	*str++ = '{';
+	*str++ = '\n';
+	*str = '\0';
+
+	return str - buf;
+}
+
+void __noprof ftrace_enter(unsigned long pc, unsigned long *lr)
```

Comment:
s/unsigned long/unsigned long int/g

Reply:
Seems like checkpatch complains for this change. Will revert to original `unsigned long`.

Reply:
sorry!